lib/auth/password_test.go

```diff
@@ -237,37 +239,54 @@ func TestServer_ChangePassword(t *testing.T) {
 	password := mfa.Password
 
 	tests := []struct {
-		name    string
-		newPass string
-		device  *TestDevice
+		name                        string
+		oldPass                     string
+		newPass                     string
+		userVerificationRequirement string
+		device                      *TestDevice
 	}{
 		{
-			name:    "OK TOTP-based change",
-			newPass: "llamasarecool11",
-			device:  mfa.TOTPDev,
+			name:                        "OK TOTP-based change",
+			oldPass:                     password,
+			newPass:                     "llamasarecool11",
+			userVerificationRequirement: "discouraged",
+			device:                      mfa.TOTPDev,
 		},
 		{
-			name:    "OK Webauthn-based change",
-			newPass: "llamasarecool13",
-			device:  mfa.WebDev,
+			name:                        "OK Webauthn-based change",
+			oldPass:                     "llamasarecool11",
+			newPass:                     "llamasarecool13",
+			userVerificationRequirement: "discouraged",
+			device:                      mfa.WebDev,
+		},
+		{
+			name:                        "OK passwordless change",
+			oldPass:                     "",
+			newPass:                     "llamasarecool15",
+			userVerificationRequirement: "required",
+			device:                      mfa.PasswordlessDev,
 		},
 	}
 
 	authServer := srv.Auth()
-	ctx := context.Background()
+	ctx := authz.ContextWithUser(
+		context.Background(),
+		authz.LocalUser{Username: username, Identity: tlsca.Identity{Username: username}},
+	)
 
-	oldPass := []byte(password)
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			oldPass := []byte(test.oldPass)
 			newPass := []byte(test.newPass)
 
 			// Acquire and solve an MFA challenge.
 			mfaChallenge, err := authServer.CreateAuthenticateChallenge(ctx, &proto.CreateAuthenticateChallengeRequest{
-				Request: &proto.CreateAuthenticateChallengeRequest_UserCredentials{
-					UserCredentials: &proto.UserCredentials{
-						Username: username,
-						Password: oldPass,
-					},
+				Request: &proto.CreateAuthenticateChallengeRequest_ContextUser{
+					ContextUser: &proto.ContextUser{},
+				},
```

Comment:
Are we changing the UI to do this as well?

I wonder if we should push this part of the request to the test table too, so we don't change the character of the tests.

Reply:
Yes, the client code changes (see `useChangePassword.ts`), and the "old way" goes away. The change there is minimal, just for compatibility sake, and the "big" change comes with the frontend PR. Unless we support third-party web clients (which, AFAIK, is not the case), I'd say we should change the test behavior here.

Reply:
I think we should keep at least one test with the old behavior, given that the functionality is still there.